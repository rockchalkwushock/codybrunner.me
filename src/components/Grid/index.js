import styled from 'styled-components'

const Grid = styled.div`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  justify-items: center;
  padding: 2em 1em;
`

Grid.displayName = 'Grid'

export default Grid
