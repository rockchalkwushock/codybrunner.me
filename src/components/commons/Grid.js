import styled from 'styled-components'
import { bool } from 'prop-types'

const Grid = styled.div`
  background-color: ${({ alternate, theme }) =>
    alternate ? theme.post.bg : 'inherit'};
  border-radius: ${({ headCard, theme }) =>
    headCard ? '0px' : theme.site.borderRadius};
  box-shadow: ${({ headCard, theme }) =>
    headCard ? 'unset' : theme.site.shadow};
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  justify-items: center;
  margin: 1em 0;
  padding: 2em 1em;
`

Grid.displayName = 'Grid'
Grid.propTypes = {
  alternate: bool,
  headCard: bool
}

export default Grid
