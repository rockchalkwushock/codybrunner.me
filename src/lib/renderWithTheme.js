import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
