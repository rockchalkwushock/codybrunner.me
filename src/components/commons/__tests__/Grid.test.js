import React from 'react'
import { shallow } from 'enzyme'

import renderWithTheme from '../../../lib/renderWithTheme'
import Grid from '../Grid'

describe('Component: <Grid />', () => {
  let alt
  let wrapper
  beforeEach(() => {
    alt = renderWithTheme(<Grid alternate />)
    wrapper = shallow(<Grid />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should render correclty as alternate', () => {
    expect(alt).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('background-color', 'inherit')
    expect(wrapper).toHaveStyleRule('display', 'grid')
    expect(wrapper).toHaveStyleRule('grid-gap', '10px')
    expect(wrapper).toHaveStyleRule('grid-template-columns', '1fr')
    expect(wrapper).toHaveStyleRule('justify-items', 'center')
    expect(wrapper).toHaveStyleRule('margin', '1em 0')
    expect(wrapper).toHaveStyleRule('padding', '2em 1em')
  })
  test('should have styles as alternate', () => {
    expect(alt).toHaveStyleRule('background-color', '#f9f8eb')
  })
})
