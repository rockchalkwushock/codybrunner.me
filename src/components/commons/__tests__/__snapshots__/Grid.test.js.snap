// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Grid /> should render correclty as alternate 1`] = `
.c0 {
  background-color: #f9f8eb;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  justify-items: center;
  margin: 1em 0;
  padding: 2em 1em;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Grid
    alternate={true}
  >
    <div
      className="c0"
    />
  </Grid>
</ThemeProvider>
`;

exports[`Component: <Grid /> should render correctly 1`] = `
.c0 {
  background-color: inherit;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  justify-items: center;
  margin: 1em 0;
  padding: 2em 1em;
}

<div
  className="c0"
/>
`;
